CMakeLists.txt

```diff
@@ -827,15 +805,18 @@ if(WITH_OPTIM)
         endif()
         if(WITH_SSE42)
             check_sse42_intrinsics()
-            if(HAVE_SSE42CRC_INLINE_ASM OR HAVE_SSE42CRC_INTRIN)
-                add_definitions(-DX86_SSE42)
-                set(SSE42_SRCS ${ARCHDIR}/adler32_sse42.c ${ARCHDIR}/insert_string_sse42.c)
-                add_feature_info(SSE42_CRC 1 "Support SSE4.2 optimized CRC hash generation, using \"${SSE42FLAG}\"")
-                list(APPEND ZLIB_ARCH_SRCS ${SSE42_SRCS})
-                set_property(SOURCE ${SSE42_SRCS} PROPERTY COMPILE_FLAGS "${SSE42FLAG} ${NOLTOFLAG}")
+            if(HAVE_SSE42CRC_INTRIN OR HAVE_SSE42CRC_INLINE_ASM)
                 if(HAVE_SSE42CRC_INTRIN)
-                    add_definitions(-DX86_SSE42_CRC_INTRIN)
+                    add_definitions(-DX86_SSE42)
```

Comment:
I'd rather see this kept as `X86_SSE42_CRC_INTRIN`, even if we only use CRC32 intrinsics from SSE42. This way it makes it more obvious that the two choices are mutually exclusive options. We can have inline assembly supported even if the processor doesn't support the instruction set.

Reply:
`X86_SSE42` means that SSE42 intrinsics in general are available, just like the similarly named defines do for all the other instruction sets.